controllers/authzctrl/reconcile_authpolicy.go

```diff
@@ -19,11 +20,16 @@ import (
 )
 
 func (r *Controller) reconcileAuthPolicy(ctx context.Context, target *unstructured.Unstructured) error {
-	desired := createAuthzPolicy(r.authComponent.Ports, r.authComponent.WorkloadSelector, r.config.ProviderName, target)
+	resolvedSelectors, err := config.ResolveSelectors(r.authComponent.WorkloadSelector, target)
+	if err != nil {
+		return fmt.Errorf("could not resolve WorkloadSelectors err: %w", err)
+	}
+
+	desired := createAuthzPolicy(r.authComponent.Ports, resolvedSelectors, r.config.ProviderName, target)
 	found := &istiosecurityv1beta1.AuthorizationPolicy{}
 	justCreated := false
 
-	err := r.Get(ctx, types.NamespacedName{
+	err = r.Get(ctx, types.NamespacedName{
```

Comment:
We are shadowing the `err` here. I wonder if we can't wrap it to `if errGet := ...; errGet != nil { ... }`. WDYT?

Reply:
like so... 

https://github.com/maistra/odh-platform/pull/62/commits/a0bc1616e75c50db5b09a37e40be40f030c76a1d